api/valor_api/schemas/evaluation.py

```diff
@@ -50,6 +50,29 @@ class EvaluationParameters(BaseModel):
     def _validate_by_task_type(cls, values):
         """Validate the IOU thresholds."""
 
+        # set default metrics for each task type
+        if values.metrics_to_return is None:
+            match values.task_type:
```

Comment:
To allow caching the evaluation should be committed to the db with all metrics that will be computed. This allows the caching i talk about in my other comment.

PrecisonRecallCurve can be an edge case handled in its own statement here